pkg/controller/perconaservermongodbbackup/perconaservermongodbbackup_controller.go

```diff
@@ -103,6 +103,18 @@ func (r *ReconcilePerconaServerMongoDBBackup) Reconcile(request reconcile.Reques
 		return rr, err
 	}
 
+	defer func() {
+		if err != nil {
+			cr.Status.State = psmdbv1.BackupStateError
+			cr.Status.Error = err.Error()
+		}
+
+		err = r.updateStatus(cr)
+		if err != nil {
+			log.Error(err, "failed to update backup status")
+		}
```

Comment:
1) I think implementation in 151-164 is better
2) should we remove 151-164 lines?

Reply:
1. Only major difference is comparing `cr.Status.State` to pbm status and IMO we shouldn't try to get status from pbm here. (Also, the logs are more verbose in 151-164 and I'll add them here too)
2. No, if we remove 151-164, we'll lose statuses other than error

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/677/commits/f83a33566690fa506cebf4f884c9aa25422e290c